composeApp/src/androidMain/kotlin/com/github/familyvault/backend/client/PrivMxClient.android.kt

```diff
@@ -149,11 +154,10 @@ class PrivMxClient : IPrivMxClient, AutoCloseable {
             UserWithPubKey(userId, publicKey)
         }
 
-        val privateMeta = if (newName != null) {
-            ThreadMetaEncoder.encode(ThreadPrivateMeta(newName, ""))
-        } else {
-            thread.privateMeta
-        }
+        val decodedPrivateMeta = ThreadMetaDecoder.decodePrivateMeta(thread.privateMeta)
+
+        val privateMeta = ThreadMetaEncoder.encode(decodedPrivateMeta.copy(name = newName ?: decodedPrivateMeta.name, threadIcon = groupChatIcon
```

Comment:
Popraw formatowanie tego

```suggestion
        val privateMeta = ThreadMetaEncoder.encode(
            decodedPrivateMeta.copy(
                name = newName ?: decodedPrivateMeta.name,
                threadIcon = groupChatIcon ?: decodedPrivateMeta.threadIcon
            )
        )
```